src/components/Table/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table'
+import { useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+const Wrapper = styled.div`
+  border: 1px solid ${({ theme }) => theme.surface3};
+  border-radius: 20px;
+  overflow: hidden;
+`
+const TableContainer = styled.div`
+  overflow: auto; // Enable horizontal scroll on full table
+`
+const TableHeadContainer = styled.div`
+  // Enable horizontal scroll on the header but hide scrollbar
+  overflow-x: auto;
+  ::-webkit-scrollbar {
+    display: none;
+  }
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+`
+const TableBodyContainer = styled.div`
+  max-height: 550px;
+  overflow-y: auto; // Enable vertical scroll only on table body
+`
+const TableSection = styled.table`
+  border-collapse: collapse;
+  box-sizing: border-box;
+`
+const TableHead = styled.thead`
+  display: table;
+  width: 100%;
+`
+const TableBody = styled.tbody`
+  display: table;
+  width: 100%;
+`
+const TableRow = styled.tr`
+  :hover {
+    background: ${({ theme }) => theme.surface3};
+  }
+`
+export const TableCell = styled.div<{ alignRight?: boolean }>`
+  width: 100%;
+  display: flex;
+  padding: 12px 16px;
+  white-space: nowrap;
+  justify-content: ${({ alignRight }) => (alignRight ? 'flex-end' : 'flex-start')};
+`
+const TableHeader = styled.th<{ width: number }>`
+  min-width: ${({ width }) => width}px;
+  padding: 8px;
+  border-bottom: 1px solid ${({ theme }) => theme.surface3};
+`
+const TableData = styled.td<{ width: number }>`
+  min-width: ${({ width }) => width}px;
+  padding: 8px;
+  box-sizing: border-box;
+`
+export function Table<Data extends object>({ columns, data }: { columns: ColumnDef<Data>[]; data: Data[] }) {
+  const headRef = useRef<HTMLTableElement>(null)
+  const bodyRef = useRef<HTMLTableElement>(null)
+  const [headerWidths, setHeaderWidths] = useState<number[]>(new Array(columns.length).fill(0))
+  const [dataWidths, setDataWidths] = useState<number[]>(new Array(columns.length).fill(0))
+
+  useLayoutEffect(() => {
+    const ths = document.querySelectorAll('thead th')
+    const tds = document.querySelectorAll('tbody tr:first-child td')
+    setHeaderWidths(Array.from(ths).map((th) => (th as HTMLElement).offsetWidth))
+    setDataWidths(Array.from(tds).map((td) => (td as HTMLElement).offsetWidth))
```

Comment:
the logic for determining width seems off as on a fullscreen macbook I believe I should be able to see all the columns?
<img width="1727" alt="Screenshot 2023-11-09 at 12 05 25 PM" src="https://github.com/Uniswap/interface/assets/11512321/6c375956-6246-4ac5-a542-5b71f98c2ab0">

And when made fullscreen, the cells still seem too large and overflow
<img width="1728" alt="Screenshot 2023-11-09 at 12 06 55 PM" src="https://github.com/Uniswap/interface/assets/11512321/82219128-dcd8-40bc-ac38-dcd179de3d9c">